src/components/Nav/Nav.js

```diff
@@ -35,10 +38,27 @@ const Nav = ({
       </div>
       <div className={styles.right}>
         {rightButtons.map((button) => {
+          const action = button.name === 'Cart' ? openCart : null;
           return (
-            <a className={styles.button} key={button.name} href={button.href}>
-              <img src={button.iconSrc} alt='' />
-              <div className={styles['button-name']}>{button.name}</div>
+            <a
```

Comment:
You should use `Link` component from `react-router-dom`, if you are using plain `a` tags your page will refresh indeed, but using `Link` will keep the browser from refreshing when going to a new URL. (React handles all the DOM updates and URL change behind the scene, so a "real" refresh is not happening)